server/persistence/dynamo/dynamo_service_player.go

```diff
@@ -0,0 +1,148 @@
+//nolint:dupl
+package dynamo
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/server/persistence"
+)
+
+const (
+	playerServiceSortKeyPrefix string = `player`
+	playerServiceGameSortKey   string = playerServiceSortKeyPrefix + `Game`
+)
+
+var _ persistence.PlayerService = (*playerService)(nil)
+
+type playerService struct {
+	ctx context.Context
+
+	svc *dynamodb.DynamoDB
+}
+
+func getPlayerService(
+	ctx context.Context,
+	svc *dynamodb.DynamoDB,
+) (persistence.PlayerService, error) {
+
+	return &playerService{
+		ctx: ctx,
+		svc: svc,
+	}, nil
+}
+
+func (ps *playerService) Get(id model.PlayerID) (model.Player, error) {
+
+	input := &dynamodb.BatchGetItemInput{
+		RequestItems: map[string]*dynamodb.KeysAndAttributes{
+			dbName: {
+				Keys: []map[string]*dynamodb.AttributeValue{{
+					partitionKey: &dynamodb.AttributeValue{
+						S: aws.String(string(id)),
+					},
+					// I want the sort key to be prefix-able
+					sortKey: &dynamodb.AttributeValue{
+						S: aws.String(string(playerServiceSortKeyPrefix)),
+					},
+					// TODO figure out what the getter should be to only nab the relevant info for the player?
+				}},
+				// TODO figure out what the projexp should be?
+				// ProjectionExpression: aws.String("max(idk)"),
+			},
+		},
+	}
+
+	dynamoResult, err := ps.svc.BatchGetItem(input)
+	if err != nil {
+		return model.Player{}, err
+	}
+	fmt.Println(dynamoResult)
+	for i, resp := range dynamoResult.Responses[dbName] {
+		dp := dynamoPlayer{}
+		// TODO unmarshalling a map doesn't work (i.e. the games)
```

Comment:
This is busted, and it looks like they don't have plans to support in v1. Might as well hop in on v2...

https://github.com/aws/aws-sdk-go/issues/3394#issuecomment-837504853